openlibrary/plugins/upstream/addbook.py

```diff
@@ -173,16 +184,24 @@ def POST(self):
             return self.no_match(saveutil, i)
 
     def find_matches(self, saveutil, i):
-        """Tries to find an edition or a work or multiple works that match the given input data.
+        """
+        Tries to find an edition or a work or multiple works that match the given input data.
 
         Case#1: No match. None is returned.
         Case#2: Work match but not editon. Work is returned.
         Case#3: Work match and edition match. Edition is returned
         Case#3A: Work match and multiple edition match. List of works is returned
         Case#4: Multiple work match. List of works is returned.
+
+        :param DocSaveHelper saveutil:
```

Comment:
`saveutil` should not be passed to `find_matches()`, which apparently can create an author and return `None`. The author creation should not occur inside a `find` method... move the author creation to `addbook.POST()` and remove `saveutil` from `find_matches()`